Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs

```diff
@@ -0,0 +1,169 @@
+    {
+        public sealed override string Name => "Toilet";
+
+        private const float PryLidTime = 1f;
+
+        private bool _isPrying = false;
+
+        [ViewVariables] public bool LidOpen { get; private set; }
+        [ViewVariables] public bool IsSeatUp { get; private set; }
+
+        [ViewVariables] private SecretStashComponent _secretStash = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _secretStash = Owner.EnsureComponent<SecretStashComponent>();
+        }
+
+        public void MapInit()
+        {
+            // roll is toilet seat will be up or down
+            var random = IoCManager.Resolve<IRobustRandom>();
+            IsSeatUp = random.NextDouble() > 0.5f;
+            UpdateSprite();
+        }
+
+        public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            // are player trying place or lift of cistern lid?
+            if (eventArgs.Using.TryGetComponent(out ToolComponent? tool)
+                && tool!.HasQuality(ToolQuality.Prying))
+            {
+                // check if someone is already prying this toilet
+                if (_isPrying)
+                    return false;
+                _isPrying = true;
+
+                if (!await tool.UseTool(eventArgs.User, Owner, PryLidTime, ToolQuality.Prying))
+                {
+                    _isPrying = false;
+                    return false;
+                }
+
+                _isPrying = false;
+
+                // all cool - toggle lid
+                LidOpen = !LidOpen;
+                UpdateSprite();
+
+                return true;
+            }
+            // maybe player trying to hide something inside cistern?
+            else if (LidOpen)
+            {
+                return _secretStash.TryHideItem(eventArgs.User, eventArgs.Using);
+            }
+
+            return false;
+        }
+
+        public bool InteractHand(InteractHandEventArgs eventArgs)
+        {
+            // trying get something from stash?
+            if (LidOpen)
+            {
+                var gotItem = _secretStash.TryGetItem(eventArgs.User);
+
+                if (gotItem)
+                    return true;
+            }
+
+            // just want to up/down seat?
+            // check that nobody seats on seat right now 
+            if (Owner.TryGetComponent(out StrapComponent? strap))
+            {
+                if (strap.BuckledEntities.Count != 0)
+                    return false;
+            }
+
+            ToggleToiletSeat();
+            return true;
+        }
+
+        public void Examine(FormattedMessage message, bool inDetailsRange)
+        {
+            if (inDetailsRange && LidOpen)
+            {
+                if (_secretStash.HasItemInside())
+                {
+                    message.AddMarkup(Loc.GetString("There is [color=darkgreen]someting[/color] inside cistern!"));
```

Comment:
`someting` in the aaaiiiir tonight
```suggestion
                    message.AddMarkup(Loc.GetString("There is [color=darkgreen]something[/color] inside cistern!"));
```